opm/material/fluidsystems/blackoilpvt/Co2GasPvt.hpp

```diff
@@ -222,11 +258,88 @@ class Co2GasPvt
         return BinaryCoeffBrineCO2::gasDiffCoeff(temperature, pressure, extrapolate);
     }
 
-    Scalar gasReferenceDensity(unsigned regionIdx) const
+    const Scalar gasReferenceDensity(unsigned regionIdx) const
     { return gasReferenceDensity_[regionIdx]; }
 
+    const Scalar oilReferenceDensity(unsigned regionIdx) const
+    { return brineReferenceDensity_[regionIdx]; }
+
+    const Scalar waterReferenceDensity(unsigned regionIdx) const
+    { return brineReferenceDensity_[regionIdx]; }
+
+    const Scalar salinity(unsigned regionIdx) const
+    { return salinity_[regionIdx]; }
```

Comment:
I didn't notice this before.  Did you mean to return a reference to `const`, i.e., `const Scalar&` here?  Otherwise there's usually very little reason to return a `const` value.